lib/DxilPIXPasses/DxilDebugInstrumentation.cpp

```diff
@@ -397,15 +402,14 @@ static unsigned FindOrAddInputSignatureElement(
                    });
 
   if (ExistingElement == InputElements.end()) {
-    auto AddedElement = llvm::make_unique<DxilSignatureElement>(sigPointKind);
+    auto AddedElement =
+        llvm::make_unique<DxilSignatureElement>(DXIL::SigPointKind::VSIn);
     unsigned int Index = static_cast<unsigned int>(InputElements.size());
-    AddedElement->Initialize(name, hlsl::CompType::getF32(),
-                             hlsl::DXIL::InterpolationMode::Constant, 1, 1,
-                             Index, 0);
-    AddedElement->AppendSemanticIndex(0);
+    AddedElement->Initialize(
+        hlsl::Semantic::Get(semanticKind)->GetName(), hlsl::CompType::getU32(),
+        hlsl::DXIL::InterpolationMode::Constant, 1, 1, Index, 0, Index, {0});
```

Comment:
I think this `FindOrAddVSInSignatureElement` function sounds more generic than it is.  The first use of `Index` in the arguments here is only correct given a particular 1:1 layout between elements in the signature and packing row, which only applies to `VSIn`.  You could assert that the `sigPointKind == DXIL::SigPointKind::VSIn` and mention why in a comment, otherwise, if this assumption is violated, you could end up with weird problems.

Reply:
Actually, it still might not be true, since it's possible to have multi-row elements on input, which could cause the Index to overlap a packing location for another element.  So, perhaps you really should iterate through the elements and find the next available row to place this element into.